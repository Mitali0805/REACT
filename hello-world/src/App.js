import React from 'react';
import logo from './logo.svg';
import './App.css';
import Greet from './Components/Greet';
import Welcome from './Components/Welcome'
import Users from './Components/Users'
import Message from './Components/Message';
import Counter from './Components/Counter';
import FunctionClick from './Components/FunctionClick';
import ClassClick from './Components/ClassClick';
import ParentComponent from './Components/ParentComponent';
import EventBind from './Components/EventBind';

function App() {
  return (
    <div className="App">
      <h1>Hello World</h1>
     <Greet/>
     <Welcome/>
     <Users title="Users List"/>
        
      <Message/> 
      <Counter/> 

      <FunctionClick/>
      <ClassClick/>

      <EventBind/>
      <ParentComponent/>
      
     </div>
  );
}

export default App;
