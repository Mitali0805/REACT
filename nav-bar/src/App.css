.App {
  text-align: center;
}


ul {
  list-style-type: none;
  margin-left: 30px;
  margin-right: 30px;
  padding: 10px;
  overflow: hidden;
  background-color:blue;
}

li{
   float: left;
   text-decoration: none;
  
}

.nav{
  display: block;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  word-spacing: 100px;
}