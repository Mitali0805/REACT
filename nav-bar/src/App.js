import React from 'react';
import './App.css';
import {BrowserRouter as Router,NavLink} from 'react-router-dom';
import Route from 'react-router-dom/Route'


function App() {
  return (
    <Router>
    <div className="App">
      
      <ul>
      <li><NavLink   className="nav" to="/" activeStyle={{color:"green"}} exact strict>Home</NavLink></li>
      <li><NavLink   className="nav" to="/projects" activeStyle={{color:"green"}} exact strict>Projects</NavLink></li>
      <li><NavLink  className="nav" to="/services" activeStyle={{color:"green"}} exact strict>Services</NavLink></li>
      <li><NavLink  className="nav" to="/contact" activeStyle={{color:"green"}} exact strict>Contact</NavLink></li>
      </ul>
      

      <Route path="/" exact strict render={
        ()=>{
          return <h1>You Chose HOME</h1>
        }
      } />

    <Route path="/projects" exact strict render={
        ()=>{
          return <h1>You Chose PROJECTS</h1>
        }
      } />

      <Route path="/services" exact strict render={
        ()=>{
          return <h1>You Chose SERVICES</h1>
        }
      } />

     <Route path="/contact" exact strict render={
        ()=>{
          return <h1>You Chose CONTACT</h1>
        }
      } />
    </div>
    </Router>
  );

  
}

export default App;
