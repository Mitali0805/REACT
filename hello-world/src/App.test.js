import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import * as renderer from 'react-test-renderer' 

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Hello World/i);
  expect(linkElement).toBeInTheDocument();
});

test('snapshot test',()=>{
  const tree = renderer.create(<App/>).toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
})