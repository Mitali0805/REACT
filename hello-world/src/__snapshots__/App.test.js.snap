// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="App"
>
  <h1>
    Hello World
  </h1>
  <h1>
    Hellloooo
  </h1>
  <h1>
    Class Component
  </h1>
  <div>
    <hr />
    example of props
    <h1>
      Users List
    </h1>
    <div>
      Name :
      Harry
      | Age:
      20
    </div>
    <div>
      Name :
      Hermaini
      | Age:
      21
    </div>
    <div>
      Name :
      Ron
      | Age:
      23
    </div>
  </div>
  <div>
    <br />
    <br />
    <hr />
    example of state
    <h1>
      Welcome to Hogwarts
    </h1>
    <button
      onClick={[Function]}
    >
      Subscribe
    </button>
  </div>
  <div>
    <br />
    <br />
    <hr />
    This is example of setState
    <h2>
      Count-
      0
    </h2>
    <button
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
  <div>
    <hr />
    example of FunctionClick
    <br />
    <button
      onClick={[Function]}
    >
      Click Button
    </button>
  </div>
  <div>
    <hr />
    example of ClassClick
    <br />
    <button
      onClick={[Function]}
    >
      Click me
    </button>
  </div>
  <div>
    <hr />
    <div>
      Hello
    </div>
    <button
      onClick={[Function]}
    >
      Click
    </button>
  </div>
  <div>
    <div>
      <hr />
      <button
        onClick={[Function]}
      >
        Greet Parent
      </button>
    </div>
  </div>
</div>
`;
